firmware/disk_drive.c

```diff
@@ -602,6 +799,10 @@ static void disk_loop(void)
 {
     D64 *d64 = &d64_state.d64;  // Reuse memory from menu
     DISK_CHANNEL *channel = 0, *channels = (DISK_CHANNEL *)crt_ram_banks[1];
+    D64_SAVE_BUFFER *save_buffer = 
+        (D64_SAVE_BUFFER *) (((uint8_t *)crt_ram_banks[1]) + sizeof(DISK_CHANNEL)*16);
```

Comment:
D64_SAVE_BUFFER will be refactored and will accommodate the following information: 
- buffered BAM (the real BAM is updated at the end of save)
- number of allocated blocks (this will be the file size)
- buffered sector data (to be written to disk)
- valid bytes in this sector data
- T-S position of the buffered sector (where to write, when full/save done)

New structure... It will be needed anyways if you'd like to emulate the 1541 buffers (#) also, later.. But I would like to leave that refactor to you when the time comes.